wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WirePlugin.kt

```diff
@@ -229,4 +256,7 @@ class WirePlugin @Inject constructor(
     } catch (e: Exception) {
       false
     }
-}
\ No newline at end of file
+}
+
+private val File.isJar
```

Comment:
I don’t think this is going to be a good approach long term. The extension might be absent depending on where the file came from.

Reply:
Could you instead optimistically treat it as a jar, and throw a nice exception if that fails?

Reply:
Hmm, but we want to avoid individual proto file paths, like this:
```
wire {
  sourcePath 'src/main/proto/squareup/geology/period.proto'
}
```
since it makes the Location.get(base, path) split logic tricky.  

We previously accomplished this by disallowing all non-directory files.  The isJar check weakens it enough since a jar is essentially a directory root.

Reply:
Ship it as-is. Later we should differentiate zip vs. non-zip by attempting to open the file as a .zip, not by its extension.